polls/tool/src/java/org/sakaiproject/poll/tool/params/PollToolBean.java

```diff
@@ -139,16 +140,17 @@ public Poll processActionAdd() {
 		boolean isNew = true;
 		if (poll.getPollId()!=null) {
 			log.debug("Actualy updating poll " + poll.getPollId());
+			Poll existingPoll = manager.getPollById(poll.getPollId(), false);
 			isNew = false;
 			//check for possible unchanged values
-			log.debug(" newPoll is " + poll.getText()+ " while poll text is " + poll.getText());
+			log.debug(" newPoll is " + poll.getText()+ " while poll text is " + existingPoll.getText());
 			
 
-			if (poll.getText().equals("") && poll.getText()!=null)
-				poll.setText(poll.getText());
-
-			if (poll.getDetails().equals("") && poll.getDetails() != null)
-				poll.setDetails(poll.getDetails());
```

Comment:
I removed these checks because they're useless as is; comparing the object's values to it's own values... Also there's already validation in place to ensure the required fields are not blank.